src/Features/Core/Portable/Snippets/SnippetProviders/AbstractConditionalExpressionAndBlockBasedSnippetProvider.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.LanguageService;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Extensions.ContextQuery;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Snippets.SnippetProviders
+{
+    /// <summary>
+    /// Base class for "if" and "while" snippet providers
+    /// </summary>
+    internal abstract class AbstractConditionExpressionAndBlockBasedSnippetProvider : AbstractSnippetProvider
```

Comment:
100% need another name here. I didn't come up with anything better than this monstrosity, so suggestions are welcome!
Note: this name needs to be synced with `AbstractCSharpConditionExpressionAndBlockBasedSnippetCompletionProviderTests`